arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/io/TaglessAdapterFactory.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.integrations.retrofit.adapter.io
+
+import arrow.Kind
+import arrow.fx.IO
+import arrow.fx.IOPartialOf
+import arrow.fx.extensions.io.async.async
+import arrow.fx.typeclasses.Async
+import arrow.integrations.retrofit.adapter.parseTypeName
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+
+class TaglessAdapterFactory<F>(private val async: Async<F>) : CallAdapter.Factory() {
+
+  companion object {
+    fun createIO(): TaglessAdapterFactory<IOPartialOf<Throwable>> = TaglessAdapterFactory(IO.async())
+
+    fun <F> create(async: Async<F>): TaglessAdapterFactory<F> = TaglessAdapterFactory(async)
+  }
+
+  override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): CallAdapter<*, *>? {
```

Comment:
nit: you can specify and narrow the return type explicitly so it's more clear for any reader and this way you don't need to specify the types for `TaglessCallAdapter` below

```suggestion
  override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): TaglessCallAdapter<F, Type>? {
```

Reply:
It will work for now, but won't work when I'll add adapter for parsing `Response<Type>`